File: Assignment 3/utilsTester.java
/**
*I Coleman and Jack Johnson
*icoleman@huskers.unl.edu / jjohnson@huskers.unl.edu
*10/14/2015
*Assignment 3, Problem 3:
*A test driver program that demonstrates 3 test calls to each of the methods created in unl.cse.Utils
*/

package unl.cse;

public class utilsTester extends Utils{
	public static void main(String args[]){
		
		try{
			System.out.println("Degrees to radians trial 1: \nInput : 1180, \nOutPut: " + Utils.degreesToRadians(1180.00));
		} catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("Degrees to radians trial 2: \nInput : -360, \nOutPut: " + Utils.degreesToRadians(-360.00));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("Degrees to radians trial 3: \nInput : 180 \nOutPut: " + Utils.degreesToRadians(180.0));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("AnnualPercentageYield trail 1: \nInput : 1\nOutput: " + Utils.annualPercentageYield(2.00));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("AnnualPercentageYield trail 2: \nInput : .5\nOutput: " + Utils.annualPercentageYield(-0.50));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("AnnualPercentageYield trail 3: \nInput : 0\nOutput: " + Utils.annualPercentageYield(0.50));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("getAirDistance trial 1: \nInput:  75 , 100, 76, 101 \nOutput: " + Utils.getAirDistance(75.0 , 100.0, 76.0, 101.0));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("getAirDistance trial 2: \nInput:  0 , 0, 0, 0 \nOutput: " + Utils.getAirDistance(10.0, -10.0, 10.0, 10.0));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}try{
			System.out.println("getAirDistance trial 3: \nInput:  175 , 10, 6, 100 \nOutput: " + Utils.getAirDistance(1750.0 , 1000.0, 6.0, 100.0));
		}catch(Exception e){
			System.out.println(e.getMessage());
		}
		
		RGB rgb = new RGB(-1, 245, 245);
		CMYK cmyk = null;
		
		try{
			cmyk = Utils.rgbToCMYK(rgb);
			System.out.println("rgbToCMYK trial 1: \nInput:  -1, 245, 245\nOutput: " + cmyk.toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}	
		
		rgb = new RGB(1, 1, 300);
		
		try{
			cmyk = Utils.rgbToCMYK(rgb);
			System.out.println("rgbToCMYK trial 2: \nInput:  1, 1, 300\nOutput: " + cmyk.toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}
		
		rgb = new RGB(200, 98345, 55);
		
		try{
			cmyk = Utils.rgbToCMYK(rgb);
			System.out.println("rgbToCMYK trial 3: \nInput:  200, 98345, 55\nOutput: " + cmyk.toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}
		
		cmyk = new CMYK(-1, 1, 1, 1);
		
		try{
			rgb = Utils.cmykToRGB(cmyk);
			System.out.println("cmykToRGB trial 1: \nInput:  -1, 1, 1, 1\nOutput: " + rgb.toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}	
		
		cmyk = new CMYK(0, 5, 1, 0);
		
		try{
			rgb = Utils.cmykToRGB(cmyk);
			System.out.println("cmykToRGB trial 2: \nInput:  0, 5, 1, 0\nOutput: " + rgb.toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}	
		
		cmyk = new CMYK(.125, .25, .95, .5);
		
		try{
			rgb = Utils.cmykToRGB(cmyk);
			System.out.println("toRGB trial 3: \nInput:  .125, .25, .95, .5\nOutput: " + rgb.toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}	
		
		rgb = new RGB(256, 255, 255);
		
		try{
			System.out.println("toGrayScale trial 1: \nInput: 256, 255, 255, LIGHTNESS\nOutput: " + Utils.toGrayScale(rgb, Mode.LIGHTNESS).toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}
		
		rgb = new RGB(-10, 1, 120);
		
		try{
			System.out.println("toGrayScale trial 2: \nInput: -10, 1, 120, LUMINOSITY\nOutput: " + Utils.toGrayScale(rgb, Mode.LUMINOSITY).toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}
		
		rgb = new RGB(5, 25, 200);
		
		try{
			System.out.println("toGrayScale trial 3: \nInput: 5, 25, 200, AVERAGE\nOutput: " + Utils.toGrayScale(rgb, Mode.AVERAGE).toString());
		}catch(Exception e){
			System.out.println(e.getMessage());
		}
	}
}

File: Assignment 3/utilsTester.c
/**
*I Coleman and Jack Johnson
*icoleman@huskers.unl.edu / jjohnson@huskers.unl.edu
*10/14/2015
*Assignment 3, Problem 1:
*A test driver program that demonstrates 3 test calls to each of the functions designed in utils.c and utils.h
*/

#include <stdio.h>
#include <stdlib.h>
#include "utils.h"
 
int main(int argc, char **argv){
	
	//declaring variables
    double radians, apy, distance, cyan, magenta, yellow, key;
    int red, green, blue;
    int error;
	Mode mode;
 
	//3 degreesToRadians tests:
    error = degreesToRadians(180, &radians);
    printf("degreesToRadians trial 1: \nInputs: %d\nRadians: %.2f, error code: %d\n",180, radians, error);
 
    error = degreesToRadians(360, &radians);
    printf("degreesToRadians trial 1: \nInputs: %d\nRadians: %.2f, error code: %d\n",360, radians, error);
 
    error = degreesToRadians(0, &radians);
    printf("degreesToRadians trial 1: \nInputs: %d\nRadians: %.2f, error code: %d\n",0, radians, error);
 
    printf("\n");
 
	//3 getAirDistance tests:
    error = getAirDistance( 0,0, 180, 180, &distance);
    printf("getAirDistance trial 1:\nInputs: %d, %d, %d, %d \nDistance: %.2f km, error code: %d\n", 0, 0, 180, 180, distance, error);
    error = getAirDistance( 0, 0, 0, 0, &distance);
    printf("getAirDistance trial 2:\nInputs: %d, %d, %d, %d \nDistance: %.2f km, error code: %d\n", 0, 0, 0, 0, distance, error);
    error = getAirDistance( 75 , 100, 76, 101, &distance);
    printf("getAirDistance trial 3:\nInputs: %d, %d, %d, %d \nDistance: %.2f km, error code: %d\n", 75 , 100, 76, 101, distance, error);
 
	//3 annualPercentageYield tests:
    printf("\n");
    error = annualPercentageYield(1, &apy);
    printf("annualPercentageYield trial 1:\nInputs: %.2f \nAPY: %.2f, error code: %d\n", 1, apy, error);
    error = annualPercentageYield(0, &apy);
    printf("annualPercentageYield trial 2:\nInputs: %.2f \nAPY: %.2f, error code: %d\n", 0, apy, error);
    error = annualPercentageYield(.5, &apy);
    printf("annualPercentageYield trial 3:\nInputs: %.2f \nAPY: %.2f, error code: %d\n", .5, apy, error);
 
	//3 rgbToCMYK tests:
    printf("\n");
    error = rgbToCMYK(1, 255, 160, &cyan, &magenta, &yellow, &key);
    printf("rgbToCMYK trial 1: \nInputs: %d, %d, %d\nCyan: %.4f, Magenta: %.4f, Yellow: %.4f, Key: %.4f error code: %d\n", 1, 255, 160, cyan, magenta, yellow, key, error);
    error = rgbToCMYK(50, 25, 10, &cyan, &magenta, &yellow, &key);
    printf("rgbToCMYK trial 2: \nInputs: %d, %d, %d\nCyan: %.4f, Magenta: %.4f, Yellow: %.4f, Key: %.4f error code: %d\n", 50, 25, 10, cyan, magenta, yellow, key, error);
    error = rgbToCMYK(255, 150, 60, &cyan, &magenta, &yellow, &key);
    printf("rgbToCMYK trial 3: \nInputs: %d, %d, %d\nCyan: %.4f, Magenta: %.4f, Yellow: %.4f, Key: %.4f error code: %d\n", 255, 150, 60, cyan, magenta, yellow, key, error);
    printf("\n");
	
	//3 cmykToRGB tests:
    error = cmykToRGB(.9, .5, .01, .1, &red, &green, &blue);
    printf("cmykToRGB trial 1: \nInputs: %.2f, %.2f, %.2f, %.2f \nRed: %d, Green: %d, Blue: %d error code: %d\n", .9, .5, .01, .1, red, green, blue, error);
    error = cmykToRGB(0, 1, 0, 1, &red, &green, &blue);
    printf("cmykToRGB trial 2: \nInputs: %.2f, %.2f, %.2f, %.2f \nRed: %d, Green: %d, Blue: %d error code: %d\n", .9, .5, .01, .1, red, green, blue, error);
    error = cmykToRGB(0, 0, 0, 0, &red, &green, &blue);
    printf("cmykToRGB trial 3: \nInputs: %.2f, %.2f, %.2f, %.2f \nRed: %d, Green: %d, Blue: %d error code: %d\n", .9, .5, .01, .1, red, green, blue, error);
    printf("\n");
	
	//3 toGrayScale tests:
	printf("\n");
	error = toGrayScale(&red, &green, &blue, AVERAGE);
	printf("toGrayScale trial 1: \nInputs: %.2f, %.2f, %.2f, %.2f \nRed: %d, Green: %d, Blue: %d, Mode: AVERAGE error code: %d\n", red, green, blue, error);
    error = toGrayScale(&red, &green, &blue, LIGHTNESS);
    printf("toGrayScale trial 2: \nInputs: %.2f, %.2f, %.2f, %.2f \nRed: %d, Green: %d, Blue: %d, Mode: LIGHTNESS error code: %d\n", red, green, blue, error);
    error = toGrayScale(&red, &green, &blue, LUMINOSITY);
    printf("toGrayScale trial 3: \nInputs: %.2f, %.2f, %.2f, %.2f \nRed: %d, Green: %d, Blue: %d, Mode: LUMINOSITY error code: %d\n", red, green, blue, error);
    printf("\n");
 
    return 0;
}